c/datasetjson.c

```diff
@@ -2251,9 +2254,238 @@ void respondWithHLQNames(HttpResponse *response, MetadataQueryCache *metadataQue
 #endif /* __ZOWE_OS_ZOS */
 }
 
+/*************************/
+#define FIELD_MAX_LENGTH    1024
+#define REQUEST_MAX_LENGTH  2048
 
-#endif /* not METTLE - the whole module */
+int qsamAllocFile( qsamFileRequest *request, char* message, 
+                   int messageLength){
+#ifdef __ZOWE_OS_ZOS
+  char *baseRequest = "alloc new catalog msg(2) ";
+  char *freeRequest = "free  ";
+  char tempString[20] = {0};
+
+  char field[FIELD_MAX_LENGTH];
+  int status, value;
+  char *ptr;
+
+  /* Must supply daName */
+  if (request->daName == NULL) {
+    return -1;
+  }
+
+  /* Allocate structure in A31 space */
+  EXTF *bpxwdyn=(EXTF *)fetch("BPXWDYN");
```

Comment:
`fetch` may return NULL. It needs to be checked.

Reply:
Thanks, fixed